File: retirement_calculator.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func getInput(prompt string, reader *bufio.Reader) int {
	fmt.Print(prompt)
	input, _ := reader.ReadString('\n')
	number, _ := strconv.Atoi(strings.TrimSpace(input))
	return number
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	currentAge := getInput("What is your current age? ", reader)
	retirementAge := getInput("At what age would you like to retire? ", reader)

	yearsLeft := retirementAge - currentAge
	year := time.Now().Year()
	retirementYear := year + yearsLeft

	fmt.Printf("You have %d years left until you can retire.\n", yearsLeft)
	fmt.Printf("It's %d, so you can retire in %d.\n", year, retirementYear)
}

File: retirement_calculator.exs
defmodule RetirementCalculator do
  def run do
    current_age = get_input("What is your current age? ")
    retirement_age = get_input("At what age would you like to retire? ")

    years_left = retirement_age - current_age
    year = Date.utc_today().year
    retirement_year = year + years_left

    response = """
    You have #{years_left} years left until you can retire.
    It's #{year}, so you can retire in #{retirement_year}.
    """

    IO.puts(response)
  end

  defp get_input(prompt) do
    prompt
    |> IO.gets()
    |> String.trim()
    |> String.to_integer()
  end
end

RetirementCalculator.run()
